atomix/utils/src/main/java/io/atomix/utils/Version.java

```diff
@@ -161,10 +173,22 @@ public static Build from(final String build) {
 
     @Override
     public int compareTo(final Build that) {
-      return ComparisonChain.start()
-          .compare(type.ordinal(), that.type.ordinal())
-          .compare(version, that.version)
-          .result();
+      ComparisonChain comparisonChain =
+          ComparisonChain.start().compare(type.ordinal(), that.type.ordinal());
+
+      // align array lengths for comparison
+      int[] left = versionIdentifiers;
+      int[] right = that.versionIdentifiers;
+      if (left.length < right.length) {
```

Comment:
:thought_balloon: Smart to use zero-padding of `Arrays.copyOf` here :+1: 
It took me a while to understand what's going on though and why this isn't broken. So maybe a comment explaining that "missing" parts of the version are substituted with zeros?